### PR TITLE
[CRT] printf/wprintf: Support %zu

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

`"%zu"` is a `printf` format specifier for type `size_t`. Some apps assume the implementation of this specifier.

JIRA issue: [CORE-17787](https://jira.reactos.org/browse/CORE-17787)

## TODO

- [x] Do small tests.
- [x] Do big tests.

## Screenshots

BEFORE:
before

AFTER:
after

### CODE REVIEW

File: sdk/lib/crt/printf/streamout.c [hunk 1]
```diff
@@ -432,6 +432,7 @@ streamout(FILE *stream, const TCHAR *format, va_list argptr)
             else if (chr == _T('w')) flags |= FLAG_WIDECHAR;
             else if (chr == _T('L')) flags |= 0; // FIXME: long double
             else if (chr == _T('F')) flags |= 0; // FIXME: what is that?
+            else if (chr == _T('z')) flags |= FLAG_INTPTR;
```
Comment (learn-more) [inline sdk/lib/crt/printf/streamout.c:435]: Should this be supported here?
Is this file not synced with wine?

Comment (katahiromz) [reply sdk/lib/crt/printf/streamout.c:435]: > Should this be supported here?

WinXP/2k3 doesn't support this. But I think it is useful for interoperability.

>Is this file not synced with wine?

File `media/doc/WINESYNC.txt`:
```txt
msvcrt -
  sdk/lib/crt/conio/cputs.c             # Synced to WineStaging-1.9.16
  sdk/lib/crt/except/cpp.c              # Synced at 20080528
  sdk/lib/crt/except/cppexcept.c        # Synced at 20071111
  sdk/lib/crt/process/_cwait.c          # Synced to WineStaging-1.7.37
  sdk/lib/crt/signal/xcptinfo.c         # Synced to WineStaging-1.7.37
  sdk/lib/crt/string/scanf.c/h          # Synced to Wine-1.7.17
  sdk/lib/crt/string/strtoi64.c         # Synced to WineStaging-1.9.9
  sdk/lib/crt/string/strtok.c           # Synced to WineStaging-1.9.16
  sdk/lib/crt/string/strtok_s.c         # Synced to WineStaging-1.9.16
  sdk/lib/crt/string/strtoul.c          # Synced to WineStaging-1.9.9
  sdk/lib/crt/string/strxfrm.c          # Synced to Wine-1.9.16
  sdk/lib/crt/string/wcs.c              # Synced at 20080611
  sdk/lib/crt/string/wctype.c           # Synced to WineStaging-1.9.16
  sdk/lib/crt/string/wtoi64.c           # Synced to Wine-1.9.16
  sdk/lib/crt/mbstring/ischira.c        # Synced to Wine-1.9.16
  sdk/lib/crt/wine/heap.c               # Synced at 20080529
  sdk/lib/crt/wine/undname.c            # Synced to WineStaging-1.9.16
  sdk/lib/crt/process/thread.c          # Synced to WineStaging-1.7.55
```

No, not synced.

**Comment by katahiromz** [comment]
Test KVM BEFORE: f73ac17 https://reactos.org/testman/compare.php?ids=85941
Test KVM AFTER: 8bec5c5
Test VBox BEFORE: f73ac17 https://reactos.org/testman/compare.php?ids=85942
Test VBox AFTER: 8bec5c5

**Comment by katahiromz** [comment]
Running `msvcrt_wintest printf` on Win10:
```txt
![image](https://user-images.githubusercontent.com/2107452/217982047-463e9ec7-de42-4d66-9fbc-b939746f75f7.png)
```

**Comment by katahiromz** [comment]
Test KVM BEFORE: f73ac17 https://reactos.org/testman/compare.php?ids=85941
Test KVM AFTER: 6b8057e
https://reactos.org/testman/compare.php?ids=85941,???
Test VBox BEFORE: f73ac17 https://reactos.org/testman/compare.php?ids=85942
Test VBox AFTER: 6b8057e
https://reactos.org/testman/compare.php?ids=85942,???

**Comment by katahiromz** [comment]
Test KVM BEFORE: f73ac17 https://reactos.org/testman/compare.php?ids=85941
Test KVM AFTER: 3ba22d3 https://reactos.org/testman/compare.php?ids=???
https://reactos.org/testman/compare.php?ids=85941,???

Test VBox BEFORE: f73ac17 https://reactos.org/testman/compare.php?ids=85942
Test VBox AFTER: 3ba22d3 https://reactos.org/testman/compare.php?ids=???
https://reactos.org/testman/compare.php?ids=85942,???

**Review by ericsomto — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]
OK for the principle; however it would be of interest to check whether Wine also supports it, because in case someone does a Winesync in the future, it would be better to not loose the functionality.

**Pull request merged**